### ISSUE TITLE
Crashes when payload is >1413 bytes

### ISSUE THREAD

**Issue by martinberlin** [IoTPanic/s#1]
I want to write this ticket to give a bit more background of the issue. This amount of bytes is on the edge of the MTU max transport unit of the ESP32. 
On the test tool using RGB I send 466 pixels (with 465 still runs)
The numbers that the testing tool reports are:

Payload:  1398   Frame size: 1413

Adding this debug line just before the memcpy in line 58 of s.cpp reports:

```c
            Serial.printf("Before memcpy: t.index:%d t.size:%d m.len:%d freeHeap:%d\n",t.index,t.size,m.len,ESP.getFreeHeap());
            memcpy(t.pyld, m.pyld, m.len);
```
Before memcpy: t.index:540 t.size:1415 m.len:1407 freeHeap:235796 (starts with 236Kb free)
            
After memcpy -> It does not die in memcpy, didn't discover yet when it does. 

But I see two issues here: 

1. The main show stopper is the payload limit when they should be stil some room more to send via UDP (Not much, last time I checked was 1470 bytes). It's not hitting the MTU limit per UDP if I see it right, and if did should not be a reason to hang, since we checked previously when that happens simply the onPacket callback is not called 

2. There is a very small but consistent memory leak that can be tracked adding this debug line. 
It starts at 235000 free Heap and every 50 seconds the heap reduces itself in 10Kb.  I was looking at it all the time today and I never caught it resetting but when the heap goes below freeHeap:58720 bytes it dies and resets itself.

But so far apart a part of this issues is working really fast so I think it has a lot of potential.

**Comment by martinberlin** [comment]
A bit more detailed update of where it dies. Not sure if it helps but at least show a more precise location.

[ s ] Got valid message
Instantiating s::transaction t
Before new uint8_t[m.transactionLength]
Before memcpy: t.index:437 t.size:1415 m.len:1407 freeHeap:200844
Was a single packet, submitting to application instead of adding to buffer
submitTransaction() t->size:1415 t->received:1415
[ s ] Got valid message  ---> There it dies just between that debug message and the IF condition afterwards

```c
// in s::receive
        #if DEBUG
        Serial.println("[ s ] Got valid message");
        #endif
        // If it isn't a recent transaction, we want to add it to our buffer
        if(!rt&&m.h.fragment==0){  --> I suspect here it dies since the next Serial output never comes
            uint16_t received = 0;

            Serial.println("Instantiating s::transaction t");
```

**Comment by IoTPanic** [comment]
Memory leak is fixed, frames that were received out of order were not being deallocated. There is still some variation in the heap availability, I did a test at 1fps and the memory did not go down at all, but when the frame rate is increased that's when the memory fluctuates. I believe this is actually allocations between the core handling the UDP requests and the core handling the callback, creating a queue of messages to handle because it's receiving so many requests.

As far as the MTU goes, keep in mind the 1470 has to include not only S data, but a UDP header, and WiFi frame information so we may be hitting the MTU. I tested it and it really does seem like we're hitting that MTU. Reopen if you would consider this a issue because @hputzek should not send packets over a specific configurable size, which if it's larger its than broken up into multiple fragments. (Obviously this is not a error on hendriks work since we have only been testing single fragments but now we can move on to multi fragments)
